sklearn/mixture/gmm.py

```diff
@@ -234,34 +386,35 @@ class GMM(BaseEstimator):
     array([ 0.5,  0.5])
 
     """
-
     def __init__(self, n_components=1, covariance_type='diag',
                  random_state=None, tol=1e-3, min_covar=1e-3,
                  n_iter=100, n_init=1, params='wmc', init_params='wmc',
                  verbose=0):
-        self.n_components = n_components
-        self.covariance_type = covariance_type
-        self.tol = tol
-        self.min_covar = min_covar
-        self.random_state = random_state
-        self.n_iter = n_iter
-        self.n_init = n_init
+        super(GMM, self).__init__(
+            n_components=n_components, covariance_type=covariance_type,
+            random_state=random_state, tol=tol, reg_covar=min_covar,
+            max_iter=n_iter, n_init=n_init, init_params=init_params,
+            verbose=verbose)
+
         self.params = params
-        self.init_params = init_params
-        self.verbose = verbose
 
-        if covariance_type not in ['spherical', 'tied', 'diag', 'full']:
-            raise ValueError('Invalid value for covariance_type: %s' %
-                             covariance_type)
+        self.weights_ = np.ones(self.n_components) / self.n_components
```

Comment:
Obviously, this line is from old ```gmm.py``` written in 2010. I don't know why this line is indicated as a new line.